docs/API/core/app-actions-v2.md

```diff
@@ -0,0 +1,460 @@
+11. Scheduled app actions will only run the published version of a screen, whereas on-demand actions will run the version from the same environment they are fired from (e.g. Fliplet Viewer, Live apps  )
+
+---
+
+## Configuring function pipeline for app actions
+
+Our latest iteration of app actions require you to define a pipeline of functions to execute when the action runs. Additionally, functions can be nested into sub-pipelines.
+
+First, you want to fetch the array of functions available in the system:
+
+```js
+Fliplet.App.Tasks.Functions.get().then(function (listOfFunctions) {
+  // ...
+});
+```
+
+Each function has the following properties:
+
+- `id`: the Fliplet ID (formerly `widgetId`) of the function
+- `name`: the display name of the function
+- `package`: the package name of the function
+- `description`: the description of the function
+
+Here'a a sample code that creates a new app action with a pipeline of function.
+
+In the example, function `com.fliplet.function.if` function that behaves like an [if statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/if...else) and evaluates the contents of the input payload and checks if `foo` is `true`. If so, it will execute function `com.example.function.return-string` with the given settings. If `foo` is `false`, the function will be skipped.
+
+```js
+Fliplet.App.Actions.create({
+  name: 'sayHello',
+  functions: [
+    {
+      functionPackage: "com.fliplet.function.if",
+      settings: {
+        condition: "foo === true"
+      },
+      functions: [
+        { functionPackage: "com.example.function.return-string", settings: { message: "Hello world" } }
+      ]
+    }
+  ]
+})
+```
+
+Each function will be executed with the configured settings. An additional payload can be added when calling the action:
+
+```js
+Fliplet.App.Actions.run('sayHello', { foo: true });
+```
+
+### Action triggers
+
+An action can be triggered as a result of a system event (e.g. a data source being updated, a log entry being created). You can configure the triggers for an action using the `triggers` property.
+
+These are the available types of triggers:
+
+- `log`: Triggered when a log entry is created. <strong>Can be executed on server side only</strong>
+- `schedule`: Executes at a specific interval. <strong>Can be executed on server side only</strong>
+- `manual`: Executes manually using the JavaScript API. <strong>Can be executed on server or client side</strong>
+- `analytics`: Triggered when an analytics event occurs. <strong>Can be executed on client side only</strong>
+
+The following example creates an action that is triggered when a log entry is created with a `type` of `dataSource.entry.create` (i.e. a new entry is created in a data source) in the data source with ID 789:
+
+```js
+Fliplet.App.Actions.create({
+  name: 'send-email-on-error',
```

Comment:
This example is also missing all the required properties for e.g environment

Reply:
@debankurghosh2061 Updated accordingly